common/plugins/org.jboss.tools.common.jdt.debug/src/org/jboss/tools/common/jdt/debug/tools/internal/Tools.java

```diff
@@ -422,6 +427,43 @@ public void invokeLoadAgent(Object virtualMachine, String path, String options)
 		}
 	}
 
+	/**
+	 * Invokes the startLocalManagementAgent method of VirtualMachine with reflection.
+	 * 
+	 * @param virtualMachine
+	 *            The virtual machine
+	 * @throws ToolsCoreException
+	 */
+	public String invokeStartLocalManagementAgent(Object virtualMachine) throws ToolsCoreException {
+		ClassLoader currentLoader = Thread.currentThread().getContextClassLoader();
+		try {
+			Thread.currentThread().setContextClassLoader(getToolsLoader());
+			Class<?> clazz = getToolsLoader().loadClass(VIRTUAL_MACHINE_CLASS);
+			Method method = clazz.getDeclaredMethod(START_LOCAL_MANAGEMENT_AGENT_METHOD, new Class[] {});
+			return (String) method.invoke(virtualMachine);
+		} catch (Throwable t) {
+			String message = t.getMessage();
+			if (message == null) {
+				Throwable cause = t.getCause();
+				while (cause != null) {
+					message = cause.getMessage();
+					if (message != null) {
+						break;
+					}
+					cause = cause.getCause();
+				}
+			}
+			throw new ToolsCoreException(IStatus.ERROR, message, t);
+		} finally {
+			Thread.currentThread().setContextClassLoader(currentLoader);
+		}
+	}
+
+	private boolean isAgentLoadExceptionWithReturnCode0(Throwable cause) {
+		return (cause.getClass().getName().contains("AgentLoadException") && "0".equals(cause.getMessage())) ||
+				(cause.getClass().equals(IOException.class) && "Non-numeric value found - int expected".equals(cause.getMessage()));
```

Comment:
I think that in the second case, we are also ignoring when there are real errors.
But in this case, it won't work anyway so we can try to improve it later and check for the _0_ return code